### PR TITLE
MINOR Allow PRs to publish build scan

### PR DESCRIPTION
**Pull request by mumrah**
Public fork pull requests triggered with `pull_request` do not get access to the repository secrets. This means they cannot upload gradle build scans. It is possible for an unprivileged `pull_request` event to publish artifacts that are downloaded by a job that has access to the secrets. https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/

### CODE REVIEW

**Comment by mumrah** [comment]
@chia7712 This one is a bit tricky, we can't fully test it until at least the ci-complete.yml workflow is merged to trunk. I tested a version of this on mumrah/kafka https://github.com/mumrah/kafka/actions/runs/10755197579 and it does seem to work. Getting build scans from PR builds will be really helpful to get more data about cache utilization, flaky tests, and build times.

WDYT about merging this one and testing it out?

File: .github/workflows/ci-complete.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: CI Complete
+
+on:
+  workflow_run:
```
Comment (chia7712) [inline .github/workflows/ci-complete.yml:19]: This approach is quiet mystery I think. Could you please add link (https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/) to it?

File: .github/workflows/ci-complete.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            false
+      - name: Setup Gradle
+        uses: ./.github/actions/setup-gradle
+        with:
+          java-version: ${{ matrix.java }}
+          develocity-access-key: ${{ secrets.GE_ACCESS_TOKEN }}
+      - uses: actions/download-artifact@v4
+        with:
+          github-token: ${{ github.token }}
+          run-id: ${{ github.event.workflow_run.id }}
```
Comment (chia7712) [inline .github/workflows/ci-complete.yml:50]: Pardon me, why we need to define `run-id`?

**Review by chia7712 — APPROVED** [review]
> WDYT about merging this one and testing it out?

yes, the comments I left can be addressed in the follow-up

File: .github/workflows/ci-complete.yml (hunk 2 above)
Comment (mumrah) [reply .github/workflows/ci-complete.yml:50]: By default, the `download-artifact` action will only download artifacts from the current workflow run. By passing `run-id` we can tell it to download artifacts from a different workflow run -- namely, the one that triggered this workflow.

**Pull request merged**